pkg/build/strategies/onbuild/onbuild.go

```diff
@@ -131,7 +146,8 @@ func (builder *OnBuild) Build(config *api.Config) (*api.Result, error) {
 
 	if len(opts.Name) > 0 {
 		if imageID, err = builder.docker.GetImageID(opts.Name); err != nil {
-			return nil, err
+			buildResult.BuildInfo = updater.UpdateFailureReason(buildResult.BuildInfo, api.ReasonGenericS2IBuildFailed)
```

Comment:
Perhaps at least a generic "docker interaction error" vs. generic s2i error (and assuming we don't want a "get image id" error).